hibernate-core/src/main/java/org/hibernate/sql/ast/spi/AbstractSqlAstTranslator.java

```diff
@@ -90,6 +101,84 @@ public void visitAssignment(Assignment assignment) {
 		return affectedTableNames;
 	}
 
+	@Override
+	public void render(SqlAstNode sqlAstNode, SqlAstNodeRenderingMode renderingMode) {
+		switch ( renderingMode ) {
+			case NO_PLAIN_PARAMETER:
+				if ( sqlAstNode instanceof SqmParameterInterpretation ) {
+					sqlAstNode = ( (SqmParameterInterpretation) sqlAstNode ).getResolvedExpression();
+				}
+				if ( sqlAstNode instanceof JdbcParameter ) {
+					final JdbcMapping jdbcMapping = ( (JdbcParameter) sqlAstNode ).getExpressionType().getJdbcMappings()
+							.get( 0 );
+					// We try to avoid inlining parameters if possible which can be done by wrapping the parameter
+					// in an expression that is semantically unnecessary e.g. numeric + 0 or concat with an empty string
+					switch ( jdbcMapping.getSqlTypeDescriptor().getJdbcTypeCode() ) {
+						case Types.BIT:
+						case Types.SMALLINT:
+						case Types.TINYINT:
+						case Types.INTEGER:
+						case Types.BIGINT:
+						case Types.DOUBLE:
+						case Types.REAL:
+						case Types.FLOAT:
+						case Types.NUMERIC:
+						case Types.DECIMAL:
+							sqlAstNode.accept( this );
+							appendSql( "+0" );
```

Comment:
you don't need to add any parens to the generated SQL?

Reply:
You are right, it's safer to add parenthesis. I'll add that.